serial_driver/test/test_serial_port.cpp

```diff
@@ -70,3 +71,48 @@ TEST(SerialPortTest, StateTest)
 
   ctx.waitForExit();
 }
+
+TEST(SerialDriverTest, SendBreakWhileClosed)
+{
+  IoContext ctx;
+  SerialPortConfig config(baud, fc, pt, sb);
+  SerialPort port(ctx, dev_null, config);
+
+  EXPECT_FALSE(port.is_open());
+  // Without the port open, expect this to fail
+  EXPECT_FALSE(port.send_break());
+
+  ctx.waitForExit();
+}
+
+TEST(SerialDriverTest, SendBreakWhenFailtToOpen)
+{
+  IoContext ctx;
+  SerialPortConfig config(baud, fc, pt, sb);
+  SerialPort port(ctx, dev_null, config);
+
+  EXPECT_FALSE(port.is_open());
+  EXPECT_THROW(port.open(), asio::system_error);
+  EXPECT_FALSE(port.is_open());
+
+  // Without the port open, should return false
+  EXPECT_FALSE(port.send_break());
+
+  ctx.waitForExit();
+}
+
+TEST(SerialDriverTest, SendBreakWhenWhileOpen)
+{
+  IoContext ctx;
+  SerialPortConfig config(baud, fc, pt, sb);
+  SerialPort port(ctx, dev_name, config);
+
+  EXPECT_FALSE(port.is_open());
+  EXPECT_NO_THROW(port.open());
```

Comment:
Not sure on your testing capability given the notes above, but I did test on physical hardware; it passes. 

I did try using a pseudo-terminal using `socat`, but it could not open the device. If Trimble decides to use this library for ROS2 serial communication. Happy to open an issue if you'd like me to look into making the tests more comprehensive.

Reply:
Thanks! I think this is fine for now. Creating a framework for testing real serial data is difficult - definitely not part of this PR either way.